span-normalizer/span-normalizer/src/main/java/org/hypertrace/core/spannormalizer/jaeger/JaegerSpanNormalizer.java

```diff
@@ -175,103 +154,69 @@ private Callable<RawSpan> getRawSpanNormalizerCallable(
     };
   }
 
-  private void sanitiseSpan(Builder rawSpanBuilder) {
-
+  // redact PII tags, tag comparisons are case-insensitive (Resource tags are skipped)
+  private void redactSpan(Builder rawSpanBuilder) {
     try {
       var attributeMap = rawSpanBuilder.getEvent().getAttributes().getAttributeMap();
-      var spanServiceName = rawSpanBuilder.getEvent().getServiceName();
+
+      Integer piiFieldsCount = 0;
+      Integer pciFieldsCount = 0;
+
       Set<String> tagKeys = attributeMap.keySet();
+      for (PIIPCIField piiPciField : PIIPCIFields) {
+        Matcher matcher = null;
+        if (piiPciField.getRegexInfo().isPresent()) {
+          matcher = piiPciField.getRegexInfo().get().getRegexPattern().matcher("");
+        }
+        PIIPCIFieldType piiPciFieldType = piiPciField.getPiiPciFieldType();
+        for (String tagKey : tagKeys) {
+          /* A Simple regex like PAN no. can have false positives.
+             Hence, we only redact tag val when tagKey is present in possible list of regexTagKeySet.
+             If regexTagKeySet is empty, we redact the tag val always.
+          */
+          if (!piiPciField.getTagKeySet().isEmpty()
```

Comment:
Move this predicate to an appositely named method.

Reply:
ok